lib/Borel/Types/Metric.hs

```diff
@@ -111,10 +113,32 @@ resourceMeasures = [Cumulative ..]
 
 prefixWeighting :: Prefix -> Double
 prefixWeighting Base = 1
-prefixWeighting Giga = 10^(9 :: Int)
-prefixWeighting Nano = 10^(-9 :: Int)
-prefixWeighting Mebi = 1024^(2 :: Int)
-prefixWeighting Mega = 10^(6 :: Int)
+prefixWeighting Giga = 10^^(9 :: Int)
+prefixWeighting Nano = 10^^(-9 :: Int)
+prefixWeighting Mebi = 1024^^(2 :: Int)
+prefixWeighting Mega = 10^^(6 :: Int)
+
+baseWeighting :: BaseUOM -> Double
+baseWeighting _ = 1
+
+weighting :: UOM -> Double
+weighting (UOM p b) = prefixWeighting p * baseWeighting b
+weighting (a `Times` b) = weighting a * weighting b
+
+extractBaseUOMs :: UOM -> MultiSet BaseUOM
+extractBaseUOMs (UOM _ b) = S.singleton b
+extractBaseUOMs (a `Times` b) = extractBaseUOMs a <> extractBaseUOMs b
+
+coercible :: UOM -> UOM -> Bool
+coercible x y = extractBaseUOMs x == extractBaseUOMs y
```

Comment:
This makes for some pretty strange logic. "giga byte times nano second" is coercible to "giga second times nano byte".
I get that all you want to do is convert the value for pretty printing, but this is still very fragile and awkward. Ideally I'd have liked the value to be uom-indexed or something, but I guess for current borel this would do.